File: common/src/main/java/io/github/mathter/memifydb/common/command/v1/XaCommitTransactionCommand.java
package io.github.mathter.memifydb.common.command.v1;

import io.github.mathter.memifydb.common.command.xa.XaCommand;

import javax.transaction.xa.Xid;

public class XaCommitTransactionCommand implements XaCommand {
    private static final byte[] PREFIX = {0x0A, 0x0D};

    private final Xid xid;

    public XaCommitTransactionCommand(Xid xid) {
        this.xid = xid;
    }

    public static byte[] getPrefix() {
        return PREFIX;
    }

    @Override
    public Xid getXid() {
        return this.xid;
    }
}

File: command/src/main/scala/io/github/mathter/memifydb/command/v1/XaCommitTransactionCommand.scala
package io.github.mathter.memifydb.command.v1

import io.github.mathter.memifydb.command.{CommandDesc, Prefix, Sequence}

import javax.transaction.xa.Xid

class XaCommitTransactionCommand(sequence: Sequence, xid: Xid, val onePhase: Boolean) extends AbstractXaCommand(sequence, xid) {
}

object XaCommitTransactionCommand extends CommandDesc {
  val prefix: Prefix = PrefixV1(Array(0x10, 0x01))
}